code/drasil-printers/Language/Drasil/Printing/Import.hs

```diff
@@ -40,19 +42,60 @@ p_space (DiscreteS a)  = "{" ++ (concat $ intersperse ", " a) ++ "}"
 parens :: P.Expr -> P.Expr
 parens = P.Fenced P.Paren P.Paren
 
-mulExpr ::  HasSymbolTable s => [Expr] -> s -> [P.Expr]
+mulExpr ::  (HasSymbolTable s, HasPrintingOptions s) => [Expr] -> s -> [P.Expr]
 mulExpr (hd1:hd2:tl) sm = case (hd1, hd2) of
   (a, Int _) ->  [expr' sm (precA Mul) a , P.MO P.Dot] ++ (mulExpr (hd2:tl) sm)
   (a, Dbl _) ->  [expr' sm (precA Mul) a , P.MO P.Dot] ++ (mulExpr (hd2:tl) sm)
   (a, _)     ->  [expr' sm (precA Mul) a , P.MO P.Mul] ++ (mulExpr (hd2:tl) sm)
 mulExpr (hd:[])      sm = [expr' sm (precA Mul) hd]
 mulExpr []       sm     = [expr' sm (precA Mul) (Int 1)]
 
+--This function takes the digits form `floatToDigits` function
+-- and decimal point position and a counter and exponent
+digitsProcess :: [Integer] -> Int -> Int -> Integer -> [P.Expr]
+digitsProcess [0] pos coun ex = [P.Int 0, P.MO P.Point, P.Int 0]
+digitsProcess (hd:tl) pos coun ex = if pos /= coun
+  then [P.Int hd] ++ (digitsProcess tl pos (coun+1) ex)
+  else if ex /= 0
+    then [P.MO P.Point, P.Int hd] ++ (map P.Int tl) ++ [P.MO P.Dot, P.Int 10, P.Sup $ P.Int ex]
+    else [P.MO P.Point, P.Int hd] ++ (map P.Int tl)
+digitsProcess [] pos coun ex = if pos > coun
+  then [P.Int 0] ++ (digitsProcess [] pos (coun+1) ex)
+  else if ex /= 0
+    then [P.MO P.Point, P.Int 0, P.MO P.Dot, P.Int 10, P.Sup $ P.Int ex]
+    else [P.MO P.Point, P.Int 0]
+
+-- THis function takes the exponent and the [Int] of base and give out
+-- the decimal point position and processed exponent
+-- This function supports transferring scientific notation to
+-- engineering notation.
+-- References for standard of Engineering Notation:
+-- https://www.khanacademy.org/science/electrical-engineering/introduction-to-ee/
+--    intro-to-ee/a/ee-numbers-in-electrical-engineering 
+-- https://www.calculatorsoup.com/calculators/math/scientific-notation-converter.php
+-- https://en.wikipedia.org/wiki/Scientific_notation
+processExpo :: Int -> (Int, Int)
+processExpo a 
+  | a == 0 = (3, -3)
+  | a == 1 = (1, 0)
+  | a == 2 = (2, 0)
+  | a == -1 = (3, -3) 
+  | a == -2 = (1, -3)
+  | a > 0 && mod (a-1)  3 == 0 = (1, a-1)
+  | a > 0 && mod (a-1)  3 == 1 = (2, a-2)
+  | a > 0 && mod (a-1)  3 == 2 = (3, a-3)
+  | a < 0 && mod (-a) 3 == 0 = (3, a-3)
+  | a < 0 && mod (-a) 3 == 1 = (2, a-2)
+  | a < 0 && mod (-a) 3 == 2 = (1, a-1)
 
 -- | expr translation function from Drasil to layout AST
-expr :: HasSymbolTable s => Expr -> s -> P.Expr
-expr (Dbl d)            _ = P.Dbl   d
-expr (Int i)            _ = P.Int   i
+expr :: (HasSymbolTable s, HasPrintingOptions s) => Expr -> s -> P.Expr
+expr (Dbl d)           sm = case sm ^. getSetting of
+  Engineering -> P.Row $ digitsProcess (map toInteger $ fst $ floatToDigits 10 d)
+     (fst $ processExpo $ snd $ floatToDigits 10 d) 0
+     (toInteger $ snd $ processExpo $ snd $ floatToDigits 10 d)
+  _           ->  P.DblSc d
```

Comment:
So if I read this code right, then in fact `DblSc` is the one which is always used, and `Dbl` is not anymore? In which case it would make sense to flip that.  So you should
1. explicitly label this case `Scientific`
2. have it output `P.Dbl`
3. change `P.Dbl` to print using scientific notation
4. remove `P.DlbSc`.

Reply:
This has been fixed in 85d4e66d2a01eb9c79c0853e362a7056d581b09e.